### PR TITLE
Revert "A warning is now issued when assertions are made for `None`"

### LINKED ISSUE #4639: Warning about assert None from XFAIL tests
Running pytest master I see warnings from XFAIL tests about asserting None. The warnings suggests to change the assert to `assert obj is None` but the reason we are asserting `None` is because the test fails (hence `XFAIL`).

I don't think these warnings should be emitted from XFAIL tests. 

Using pytest master:
```python
import pytest

@pytest.mark.XFAIL
def test_f():
    assert g()

def g():
    # Shouldn't return None but doesn't work properly...
    return None
```

Running the tests gives:
```console
$ pytest test_f.py 
====================================================== test session starts =======================================================
platform darwin -- Python 2.7.10, pytest-4.1.1, py-1.7.0, pluggy-0.8.0
rootdir: /Users/enojb/current/sympy/pytest, inifile: tox.ini
plugins: xdist-1.26.0, forked-0.2, doctestplus-0.3.0.dev0
collected 1 item                                                                                                                 

test_f.py F                                                                                                                [100%]

============================================================ FAILURES ============================================================
_____________________________________________________________ test_f _____________________________________________________________

    @pytest.mark.XFAIL
    def test_f():
>       assert g()
E       PytestWarning: asserting the value None, please use "assert is None"

test_f.py:5: PytestWarning
==================================================== short test summary info =====================================================
FAIL test_f.py::test_f
==================================================== 1 failed in 0.08 seconds ====================================================
```

### PR DESCRIPTION
**Pull request by asottile**
Resolves #4639 

CC @Tadaboody

### CODE REVIEW

File: testing/test_warnings.py [hunk 1]
```diff
@@ -543,7 +543,7 @@ def assert_result_warns(result, msg):
 
     def test_tuple_warning(self, testdir):
         testdir.makepyfile(
-            """
+            """\
```
Comment (blueyed) [inline testing/test_warnings.py]: Please do not change code unnecessary, especially when it is only to please your editor in some way (IIRC).
For all others this adds visual disbalance, and often gets highlighted in a special way (line continuation).

**Comment by nicoddemus** [comment]
This can go to `master` btw (it can't possibly break anything).

**Comment by asottile** [comment]
rebased on master!

File: testing/test_warnings.py (hunk 1 above)
Comment (blueyed) [reply testing/test_warnings.py]: ?

**Comment by Tadaboody** [comment]
Oof another one bites the dust then (honestly it’s pretty satisfying seeing the revert being this smooth).

Is it worth it to try and eliminate the false positive? Maybe try and only warn if it fails the test? (Some kinda `try` wrapping it)

**Comment by nicoddemus** [comment]
@Tadaboody I don't think so, most cases the test will at least fail with a message anyway, as @asottile pointed out.

@blueyed anything else here?

**Comment by blueyed** [comment]
> @blueyed anything else here?

I do not like the (unnecessary) code style change (and being ignored for pointing it out (across a rebase where it could have been fixed)), and will likely investigate in what I've said in https://github.com/pytest-dev/pytest/issues/4639#issuecomment-555845784.
I am not blocking this though, but will likely carry this into my fork then also.. :+1:

**Pull request merged**